pkg/ccl/backupccl/backup_test.go

```diff
@@ -8931,56 +8818,64 @@ func TestBackupOnlyPublicIndexes(t *testing.T) {
 		// Take an incremental backup and assert that it doesn't contain any
 		// data. The only added data was from the backfill, which should not
 		// be included because they are historical writes.
-		_, err := sqlDB.DB.ExecContext(ctx, `BACKUP DATABASE data TO $1 INCREMENTAL FROM $2 WITH revision_history`,
-			inc1Loc, fullBackup)
+		sqlDB.QueryRow(t, "SELECT cluster_logical_timestamp()").Scan(&inc1Ts)
+		_, err := sqlDB.DB.ExecContext(ctx,
+			fmt.Sprintf(
+				`BACKUP DATABASE data INTO $1 AS OF SYSTEM TIME %s WITH revision_history, incremental_location=$2`,
+				inc1Ts,
+			),
+			fullBackup, incLoc)
 		return errors.Wrap(err, "running inc 1 backup")
 	})
 
 	g.Go(func() error {
 		defer backfillBlockers[1].allowToProceed()
 		backfillBlockers[1].waitUntilBlocked()
 
+		sqlDB.QueryRow(t, "SELECT cluster_logical_timestamp()").Scan(&inc2Ts)
 		// Take an incremental backup and assert that it doesn't contain any
-		// data. The only added data was from the backfill, which should not be
-		// included because they are historical writes.
-		_, err := sqlDB.DB.ExecContext(ctx, `BACKUP DATABASE data TO $1 INCREMENTAL FROM $2, $3 WITH revision_history`,
-			inc2Loc, fullBackup, inc1Loc)
+		// data. The only added data was from the backfill, which should not
+		// be included because they are historical writes.
+		_, err := sqlDB.DB.ExecContext(ctx,
+			fmt.Sprintf(
+				`BACKUP DATABASE data INTO $1 AS OF SYSTEM TIME %s WITH revision_history, incremental_location=$2`,
+				inc2Ts,
+			),
+			fullBackup, incLoc)
 		return errors.Wrap(err, "running inc 2 backup")
 	})
 
 	// Wait for the backfill and incremental backup to complete.
 	require.NoError(t, g.Wait())
 
-	inc1Spans := getSpansFromManifest(ctx, t, locationToDir(inc1Loc))
-	require.Equalf(t, 0, len(inc1Spans), "expected inc1 to not have any data, found %v", inc1Spans)
-
-	inc2Spans := getSpansFromManifest(ctx, t, locationToDir(inc2Loc))
-	require.Equalf(t, 0, len(inc2Spans), "expected inc2 to not have any data, found %v", inc2Spans)
-
 	// Take another incremental backup that should only contain the newly added
 	// index.
-	inc3Loc := localFoo + "/inc3"
-	sqlDB.Exec(t, `BACKUP DATABASE data TO $1 INCREMENTAL FROM $2, $3, $4 WITH revision_history`,
-		inc3Loc, fullBackup, inc1Loc, inc2Loc)
-	inc3Spans := getSpansFromManifest(ctx, t, locationToDir(inc3Loc))
-	require.Equal(t, 1, len(inc3Spans))
-	require.Regexp(t, fmt.Sprintf(".*/Table/%d/{2-3}", dataBankTableID), inc3Spans[0].String())
```

Comment:
Ended up removing these checks — trying to get the paths of the incrementals is a bit of a pain since there's no SQL query we can make to get their paths unlike `SHOW BACKUPS IN` for fulls.

I don't think they're too necessary anyway, given that the subtests below perform restores and check for the existence/non-existence of the indices.